### PR TITLE
Add SiYuan icon

### PR DESCRIPTION
**Pull request by Zuoqiu-Yingyi**
image

**GitHub stars:** [~13.7k (2023-12-26)](https://github.com/siyuan-note/siyuan)
**Similarweb rank:** [~516k (2023-12-26)](https://www.similarweb.com/website/b3log.org/)

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

- HEX value source: [Brand Marking Color table #d23f31](https://b3log.org/brand-marking.html#:~:text=%C2%A0%20%233b3e43-,%E7%BA%A2%E8%89%B2%20%C2%A0%20%23d23f31,-%E8%93%9D%E8%89%B2%20%C2%A0%20%234285f4)
- manually vectorize the logo

### CODE REVIEW

**Comment by Zuoqiu-Yingyi** [comment]
It seems that the JSON Schema verify of `aliases.loc` property is not compatible with `BCP 47` tag  contains `Script` subtag such as `zh-Hans` and `zh-Hant` 👀?

https://github.com/simple-icons/simple-icons/blob/6f977196866be0f3895d73371086a49396253251/.jsonschema.json#L651-L663

REF: [RFC 5646: Script Subtag](https://www.rfc-editor.org/rfc/rfc5646#section-2.2.3)
REF: https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry

**Comment by adamrusted** [comment]
> It seems that the JSON Schema verify of `aliases.loc` property is not compatible with `BCP 47` tag contains `Script` subtag such as `zh-Hans` and `zh-Hant` 👀?
> 
> https://github.com/simple-icons/simple-icons/blob/6f977196866be0f3895d73371086a49396253251/.jsonschema.json#L651-L663
> 
> REF: [RFC 5646: Script Subtag](https://www.rfc-editor.org/rfc/rfc5646#section-2.2.3) REF: [iana.org/assignments/language-subtag-registry/language-subtag-registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry)

Correct, the `aliases.loc` uses ISO 639 (the original version) for our language tags. As of right now, we don't differentiate between `zh-Hans` and `zh-Hant` - instead linking by a region, though we do only have [this](https://github.com/simple-icons/simple-icons/blob/ff48cf3e301193f523a022549e2bcfdf6576d75d/_data/simple-icons.json#L8689-L8698) and [this](https://github.com/simple-icons/simple-icons/blob/ff48cf3e301193f523a022549e2bcfdf6576d75d/_data/simple-icons.json#L9507-L9516) as examples within our codebase. If you could adapt this PR to suit ISO 639 for now, and perhaps open an issue up to discuss changing our language codes used - we can look to review and get this merged in.

**Comment by Zuoqiu-Yingyi** [comment]
@adamrusted
To Be Reviewed.
This PR is currently `ISO 639` adapted.

File: _data/simple-icons.json [hunk 1]
```diff
@@ -13086,6 +13086,24 @@
             "hex": "258AAF",
             "source": "https://www.sitepoint.com"
         },
+        {
+            "title": "SiYuan",
+            "hex": "D23F31",
+            "source": "https://b3log.org/brand-marking.html",
```
Comment (adamrusted) [inline _data/simple-icons.json:13092]: Did you generate this from the PNG, or from the SVG in the site header? If the latter, we should change the source to reflect that.

```suggestion
            "source": "https://b3log.org/siyuan",
```

File: icons/siyuan.svg
Comment (adamrusted) [inline icons/siyuan.svg:1]: Icon itself is a very clean version of what I'd expect to see for this brand, well done @Zuoqiu-Yingyi 👏🏻

**Review by adamrusted — COMMENTED** [review]
I'm personally happy with the icon itself, given how _rubbish_ the source SVG is!
Given you've done a fair bit of work on this though, I'll ask another maintainer to review before we merge!

File: _data/simple-icons.json (hunk 1 above)
Comment (Zuoqiu-Yingyi) [reply _data/simple-icons.json:13092]: The original version of this icon was a bitmap image (`*.psd`), the SVG version is converted from bitmap. So I think `https://b3log.org/brand-marking.html` is a better than `https://b3log.org/siyuan`.

Comment (adamrusted) [reply _data/simple-icons.json:13092]: I can only find `.png` files on that page, am I missing something?

Comment (Zuoqiu-Yingyi) [reply _data/simple-icons.json:13092]: > I can only find `.png` files on that page, am I missing(失踪) something?

Oh, I'm one of the core contributors to this open source project, so I have a undisclosed draft of this icon.

**Comment by PeterShaggyNoble** [comment]
Merging this in based on Adam's approval above.

**Pull request merged**